interp/run.go

```diff
@@ -3043,13 +3049,23 @@ func _case(n *node) {
 					}
 					return fnext
 				}
-				if v := val.Interface().(valueInterface).node; v != nil {
-					for _, typ := range types {
-						if v.typ.id() == typ.id() {
-							destValue(f).Set(val)
-							return tnext
+				if vi, ok := val.Interface().(valueInterface); ok {
+					if v := vi.node; v != nil {
```

Comment:
why do we not have here first the equivalent of 3004-3009 ? i.e. when vi.node == nil and typ.cat == nitT ?